From c84f4e0039353e3846afda132a53d96fcadc0715 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Sun, 30 Nov 2014 11:01:00 -0800
Subject: [PATCH] application: use correct order to watch for GSettings changes

After a recent GLib change, we need to connect to the changed signal
before we actually read its value, or we will never get a change
notification.
---
 src/nemo-application.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/nemo-application.c b/src/nemo-application.c
index 346fa3d..54f4726 100644
--- a/src/nemo-application.c
+++ b/src/nemo-application.c
@@ -1077,6 +1077,11 @@
 				  G_CALLBACK (desktop_changed_callback),
 				  self);
 
+	g_signal_connect_swapped (gnome_background_preferences, "changed",
+				  G_CALLBACK (desktop_changed_callback),
+				  self);
+	desktop_changed_callback (self);
+
 	g_signal_connect (screen, "monitors-changed",
 				  G_CALLBACK (monitors_changed_callback),
 				  self);

